### PR TITLE
MUIC-565: [Android] Disable overlay permission dialog (if bubble disabled)

### PR DESCRIPTION
**Pull request by deniss-mc**
**Jira issue:**
https://glia.atlassian.net/browse/MUIC-565

**Additional info**
Will use `GliaConfigurationManager` for Configuration holder (will need to move whitelabel stuff there too)
At some point would love to introduce some sort of public interface with all the non-ui related configurations (or maybe add some to `GliaConfig` class - some will come from the core-sdk "purchased package" anyway in the future.

### CODE REVIEW

**Comment by dukhovnyi** [comment]
could we introduce this kind of setting in our TestingApp also to have possibility to test it by QAs?

**Review by dukhovnyi — COMMENTED** [review]
could we introduce this kind of setting in our TestingApp also to have possibility to test it by QAs?

**Review by gugalo — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         queueId = intent.getStringExtra(GliaWidgets.QUEUE_ID);
         runtimeTheme = intent.getParcelableExtra(GliaWidgets.UI_THEME);
         contextUrl = intent.getStringExtra(GliaWidgets.CONTEXT_URL);
-        useOverlays = intent.getBooleanExtra(GliaWidgets.USE_OVERLAY, true);
+        useOverlays = intent.getBooleanExtra(GliaWidgets.USE_OVERLAY, Dependencies.getSdkConfigurationManager().isUseOverlay());
```
Comment (jakob-mc) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java:71]: I suggest to use a separate method to get default value, improves readability.

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.java [hunk 2]
```diff
@@ -201,6 +201,7 @@ public void startCall(String companyName,
                           String contextUrl,
                           boolean useOverlays,
                           Engagement.MediaType mediaType) {
+        Dependencies.getSdkConfigurationManager().setUseOverlay(useOverlays);
```
Comment (jakob-mc) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallView.java:204]: `startCall` method is doing many different things here: setting configuration, initCall, onNavigatedToCall. Method could be broken down into multiple methods with less responsibilities.

Comment (jakob-mc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallView.java:204]: Is there any more appropriate layer (presenter, controller etc) to put that kind of code?

**Review by jakob-mc — CHANGES_REQUESTED** [review]

**Comment by deniss-mc** [comment]
> could we introduce this kind of setting in our TestingApp also to have possibility to test it by QAs?

Added `Use Overlay` option to settings (enabled by default)- 
To test it:
1) Wipe data
2) Disable in settings
3) Go to ChatView

Expected:
Overlay Permissions dialog is not displayed

File: app/src/main/java/com/glia/exampleapp/MainFragment.java [hunk 3]
```diff
@@ -80,6 +80,6 @@ private void setNavigationIntentData(Intent intent, SharedPreferences sharedPref
         UiTheme uiTheme = Utils.getUiThemeByPrefs(sharedPreferences, getResources());
         intent.putExtra(GliaWidgets.UI_THEME, uiTheme);
         // use to set the bubble functionality
-        // intent.putExtra(GliaWidgets.USE_OVERLAY, false);
```
Comment (jakob-mc) [inline app/src/main/java/com/glia/exampleapp/MainFragment.java:83]: cleanup

**Review by dukhovnyi — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java (hunk 1 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java]: Introduced Configuration class - moved stuff around a bit

**Review by jakob-mc — CHANGES_REQUESTED** [review]

**Review by jakob-mc — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.java (hunk 2 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallView.java:204]: Prefer not to do it atm. 
- breaking down is not related to this commit (will add some sort of technical debt task for that)
- its extra usecase for setting one variable

Atm. Views public interface requires variables

Will refac to using Configuration class ASAP (check PR notes)

**Pull request merged**